### PR TITLE
feat: Hook up DELETE endpoint in API.

### PR DESCRIPTION
**Pull request by atomdmac**
Sends deletion requests to the API for user rankings.

**NOTE:**  There is currently a UNIQUE constraint on the `UserRankings.prev_ranking_id` that prevents items from being ranked after they have been un-ranked.  This is due to the fact that we're currently soft-deleting records in the table.  The easiest solution is to switch back over to hard deletes but I'm open to other suggestions.

@SharpNotions/ten-hour-project

### CODE REVIEW

**Review by payers1 — APPROVED** [review]
I'm open to going back to hard deletes, since it's probably more important to delete a single item rank than to clear all rankings. I couldn't find a good way to clear all rankings without soft delete, but I'm sure someone smarter could find a good solution

**Comment by atomdmac** [comment]
I also read (very briefly) about using Postgres's `EXCLUDE` constraint to mimic a `UNIQUE` constraint that omits soft-deleted records.  I didn't spend enough time with it to determine for sure if that would suit our use case here, though.

**Comment by ryanSN** [comment]
I wouldn't do the exclude, as it will make querying later on a pain. Would there be a reason for the soft delete on a user ranking, I can see it on items. If we are looking for a history of what the user added and where I would suggest lets do [audit triggers](https://github.com/2ndQuadrant/audit-trigger/blob/master/audit.sql).

tl;dr lets do hard deletes
